extension/InspectionDoc/src/Infrastructure/Repository/InspectionDocRepository.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace InspectionDoc\Infrastructure\Repository;
+
+use InspectionDoc\Entity\InspectionDoc;
+use InspectionDoc\Entity\InspectionDocInterface;
+use InspectionDoc\Infrastructure\DataProvider\InspectionDocDataProviderInterface;
+
+class InspectionDocRepository implements InspectionDocRepositoryInterface
+{
+    /**
+     * @return array<string, array<mixed>>
+     */
+    protected ?array $indexedInspectionDocs = null;
+
+    /**
+     * @var array<string, \InspectionDoc\Entity\InspectionDocInterface>
+     */
+    protected array $inspectionDocsCache = [];
+
+    /**
+     * @var \InspectionDoc\Infrastructure\DataProvider\InspectionDocDataProviderInterface
+     */
+    protected InspectionDocDataProviderInterface $inspectionDocDataProvider;
+
+    /**
+     * @param \InspectionDoc\Infrastructure\DataProvider\InspectionDocDataProviderInterface $inspectionDocDataProvider
+     */
+    public function __construct(InspectionDocDataProviderInterface $inspectionDocDataProvider)
+    {
+        $this->inspectionDocDataProvider = $inspectionDocDataProvider;
+    }
+
+    /**
+     * @param string $errorCode
+     *
+     * @return \InspectionDoc\Entity\InspectionDocInterface|null
+     */
+    public function findByErrorCode(string $errorCode): ?InspectionDocInterface
+    {
+        $indexedInspectionDocs = $this->getIndexedInspectionDocs();
+
+        $inspectionId = $this->findInspectionIdByErrorCode(array_keys($indexedInspectionDocs), $errorCode);
+
+        if ($inspectionId === null) {
+            return null;
+        }
+
+        $inspectionDoc = $indexedInspectionDocs[$inspectionId];
+
+        return $this->createInspectionDoc($inspectionId, $inspectionDoc);
+    }
+
+    /**
+     * @param array $inspectionsIds
+     * @param string $errorCode
+     *
+     * @return string|null
+     */
+    protected function findInspectionIdByErrorCode(array $inspectionsIds, string $errorCode): ?string
+    {
+        $inspectionIdCandidates = [];
+
+        foreach ($inspectionsIds as $inspectionsId) {
+            if (!$this->strStartsWith($errorCode, $inspectionsId)) {
+                continue;
+            }
+
+            $inspectionIdCandidates[] = $inspectionsId;
+        }
+
+        return $this->getMaxLengthMatch($inspectionIdCandidates);
+    }
+
+    /**
+     * @param string $haystack
+     * @param string $needle
+     *
+     * @return bool
+     */
+    protected function strStartsWith(string $haystack, string $needle): bool
```

Comment:
In my opinion, it's a good candidate for a static method in some helper/util class.

Reply:
it's just one operation. we don't need to create a helper for that

Reply:
one function, that can be used statically and has no dependency from repository, where it is located. I've heart that people in Spryker don't like static methods, so if you want, leave it here.